src/frontend/src/optimizer/plan_node/logical_project.rs

```diff
@@ -305,6 +306,14 @@ impl ToStream for LogicalProject {
         let (input, input_col_change) = self.input().logical_rewrite_for_stream(ctx)?;
         let (proj, out_col_change) = self.rewrite_with_input(input.clone(), input_col_change);
 
+        if proj.exprs().iter().any(|expr| expr.count_nows() > 0) {
```

Comment:
I think this is quite hacky? Now can be placed in almost any location including order by expression. The way we handle it currently is not so good. Perhaps we can try to ban binding of now if it is not in a where clause?

Reply:
> I think this is quite hacky? Now can be placed in almost any location including order by expression. The way we handle it currently is not so good. Perhaps we can try to ban binding of now if it is not in a where clause?

but batch needs to bind `now` outside where clause..

Reply:
Yes, seems there are two solutions:
1. Bind now as valid expression only in batch
2. Binder will reject now in any location but WHERE clause (i.e. filter) when creating stream

Both require `Binder` to be aware of whether it is binding from batch or stream.

Reply:
> Yes, seems there are two solutions:
> 
> 1. Bind now as valid expression with the `session_timestamp` input only in batch
> 2. Binder will reject now in any location but WHERE clause (i.e. filter) when creating stream
> 
> Both require `Binder` to be aware of whether it is binding from batch or stream. But this is trivial, we can have e.g. `Binder::new_for_stream` when handling create MV.
> 
> I prefer the latter solution as we do not have to wait for the compute node to try to create an executor for it to fail.We already intercept the `now` expression in `WHERE` clauses...
> 
> Actually, I prefer if _both_ solutions are implemented. This way, `now` should have arity 1 in batch mode and arity 0 in stream mode. Why not you work on the 1. and I can PR in something for 2.

thanks. I have just done 1.